lib/legacy-logging.js

```diff
@@ -0,0 +1,764 @@
+    entry.level = 'all'
+  }
+
+  if (!entry.handler) {
+    entry.handler = handlers.print
+  }
+
+  var routes = []
+
+  _.each(propnames, function (pn) {
+    var valspec = entry[pn]
+
+    if (valspec) {
+      // vals can be separated by either comma or space, comma takes precedence
+      // spaces are useful for command line, as comma is used up
+      var vals = valspec.replace(/\s+/g, ' ').split(/[, ]/)
+      _.map(vals, function (val) { return val.replace(/\s+/g, '') })
+      vals = _.filter(vals, function (val) { return val !== '' })
+
+      if (pn === 'level') {
+        var newvals = []
+        _.each(vals, function (val) {
+          if (val === 'all') {
+            newvals = newvals.concat(loglevels)
+          }
+          else if (val.match(/\+$/)) {
+            val = val.substring(0, val.length - 1).toLowerCase()
+            newvals = newvals.concat(loglevels.slice(loglevels.indexOf(val)))
+          }
+          else {
+            newvals.push(val.toLowerCase())
+          }
+        })
+
+        vals = _.uniq(newvals)
+        _.each(vals, function (level) {
+          if (loglevels.indexOf(level) === -1) {
+            throw error('invalid_log_level', {level: level})
+          }
+        })
+      }
+
+      var newroutes = []
+
+      _.each(vals, function (val) {
+        if (!routes.length) {
+          var newroute = {}
+          newroute[pn] = val
+          newroutes.push(newroute)
+        }
+        else {
+          _.each(routes, function (route) {
+            var newroute = Common.copydata(route)
+            newroute[pn] = val
+            newroutes.push(newroute)
+          })
+        }
+      })
+
+      routes = newroutes
+    }
+  })
+
+  _.each(routes, function (route) {
+    var routestr = Util.inspect(route)
+
+    var handler = entry.handler
+
+    if (handler === 'print') {
+      handler = handlers.print
+    }
+
+    if (!_.isFunction(handler)) {
+      throw error('handler_not_function', { entry: entry })
+    }
+
+    if (handler) {
+      handler.routestr = routestr
+    }
+
+    // must match exact route
+    var prev = logrouter.findexact(route)
+
+    if (!handler) {
+      if (prev) {
+        var remove = true
+        if (prev.multiplex) {
+          // FIX: this doesn't really work - could pop anything
+          prev.multiplex.pop()
+          remove = !prev.multiplex.length
+        }
+        if (remove) {
+          logrouter.remove(route)
+        }
+      }
+    }
+    else {
+      if (prev) {
+        handler = multiplexhandler(prev, entry.handler)
+        handler.routestr = routestr
+      }
+
+      if (entry.regex) {
+        handler = make_regex_handler(entry.regex, handler)
+      }
+
+      else if (entry.act) {
+        handler = make_act_handler(entry.act, handler)
+      }
+
+      else if (entry.pin) {
+        handler = make_pin_handler(entry.pin, handler)
+      }
+
+      logrouter.add(route, handler)
+    }
+  })
+}
+
+/*
+
+logspec.map:
```

Comment:
When broken out into it's own plug, can we move this into an MD file. Otherwise it will never be read.